src/decode/lzma.rs

```diff
@@ -239,48 +197,53 @@ where
     Ok(decoder)
 }
 
-impl<W, LZB> DecoderState<W, LZB>
-where
-    W: io::Write,
-    LZB: lzbuffer::LzBuffer<W>,
-{
+impl DecoderState {
     pub fn reset_state(&mut self, lc: u32, lp: u32, pb: u32) {
+        if self.lc + self.lp == lc + lp {
+            // we can just reset here..
+            self.literal_probs.iter_mut().for_each(|v| v.fill(0x400))
```

Comment:
This requires Rust 1.50+. Can you increase the MSRV as such in https://github.com/gendx/lzma-rs/blob/master/.github/workflows/tests.yml#L14 and https://github.com/gendx/lzma-rs/blob/master/README.md?